src/Main.hs

```diff
@@ -32,16 +33,29 @@ parse _ = exit
 main :: IO ()
 main = do
     args <- getArgs
+    logR <- initLogger
+    let logMsg = logMessage logR
+    logMsg $ "Parsing input file: " ++ concat args
     torrentStr <- parse args
     case decode torrentStr of
       Right d ->
           case mkMetaInfo d of
-            Nothing -> putStrLn "parse error"
+            Nothing -> logMsg "parse error"
             Just m -> do
+              logMsg "Input File OK"
```

Comment:
Should be shell buffering i guess? Can you try again with `hSetBuffering stdout NoBuffering` please?